src/ast.js

```diff
@@ -30,26 +30,31 @@ class Alternation {
  * Represents a character class (e.g. [0-9a-z])
  */
 class CharClass {
-  constructor(a, b) {
+  constructor(a, b, flags) {
     this.precedence = 1;
     this.set = regenerate(a, b);
+    this.flags = flags;
   }
 
   get length() {
     return 1;
   }
 
+  get isSingleCharacter() {
+    return !this.set.toArray().some(c => c > 0xffff);
```

Comment:
I ended up adding an `isSingleCodepoint` getter to these two, since the context of the flags has been moved to only be in `toString` and, in this case, `parens` - that way the unicode context doesn’t need to be passed to these particular methods.